CMakeLists.txt

```diff
@@ -67,7 +67,13 @@ ENDFUNCTION()
 
 LIST(APPEND CMAKE_MODULE_PATH cmake/Modules)
 
-IF(NOT KOKKOS_HAS_TRILINOS)
+
+IF(KOKKOS_HAS_TRILINOS)
+  # What language are we compiling Kokkos as
+  # downstream dependencies need to match this!
+  # We do not supporting using CUDA or HIP as cmake language yet in Trilinos
+  SET(KOKKOS_COMPILE_LANGUAGE CXX)
```

Comment:
It looks like this appears in both the `IF` and `ELSE` branches - why not lift it up above?

Reply:
Actually it gets reset again in the else branch if you have CUDA on, I thought its cleaner if the Trilinos path makes clear we always set it to CXX and it doesn't get messed with later again.

Reply:
Works for me